FixMyBerlin/fixmy.frontend
# Visualize links between reports and plannings

Bike stands reports and plannings can be linked when a planning implements reports or when a report is implemented by a planning. These links have been displayed as a list in the detail side panel for entries up until now. However, it is difficult for users to understand the location of linked entries without a visual aid.

This pull request implements a visualisation of links between reports and plannings by adding lines to the map connecting them whenever an entry is selected or hovered over. An entry is considered selected when its marker pin has been clicked or when its direct URL has been navigated to. 

Closes #568

# Configuration

The link visualisation can be enabled or disabled by adjusting a region's reports configuration. If the key `config.reports.overviewMap.linkLayer` is not set, no links are displayed on the map.

The color of links can be configured by setting `config.reports.overviewMap.linkLayer.color`.

# Caveats

- The source and destination of link doesn't match marker pins exactly because marker pins are not displayed with their center on the location of an entry
- Links can point at empty spots on the map when source or destionation are displayed as marker clusters

# Testing

Test this feature by:

* set `REGION=aachen` in .env, run dev server
* open http://localhost:8080/meldungen/karte/46. Linkages should be visualized
* hover other markers in the map, inspect their linkages

# Todo

- [x] Consider refactoring arc service using component composition. I would like to contain the arc functionality instead of having it spread over a handful of modules.
- [x] Viewpoint shift when detail panel is open
- [ ] UX check
- [ ] IE11 compatibility testing
